### ISSUE TITLE
Weird behavior with LUAJIT_SECURITY_PRNG

### ISSUE THREAD

**Issue by rfl890** [LuaJIT/LuaJIT#995]
Hi,
When compiling LuaJIT on Windows using MSVC (without changing anything in the header files), running `math.random()` always gives the same result, 0.79420629243124.  I have added a printf at the start of the `lj_prng_seed_secure` (when `LUAJIT_SECURITY_PRNG` is 1), and when I run the interpreter, it prints to the console! Also, when changing the seed in the `lj_prng_seed_fixed`, it also changes the random output. Is this intended behavior or an issue on my end?

**Comment by vanc** [comment]
Shouldn't you seed the random number generator first?

```
math.randomseed(os.time())
```

**Comment by MikePall** [comment]
The secure PRNG is only used for VM internals. It's randomly seeded. It's state must not be directly exposed to applications.

But it has nothing do to with the PRNG for `math.random()`, except the underlying PRNG algorithm is the same. `math.random()` in LuaJIT has to follow the (very limited) Lua API. It starts from a deterministic seed and it's **not** secure. No matter how you seed it.

Quoting the docs:
| Important: Neither this nor any other PRNG based on the simplistic `math.random()` API is suitable for cryptographic use.

tl;dr: Use `math.random()` for some toy randomness, e.g. in games. If you want different results on every startup, then you need to seed it,  see above.
**Never** use this API for security-related purposes, e.g. to generate nonces, keys, access tokens, UUIDs and so on.
